src/go/rpk/pkg/cli/acl/acl.go

```diff
@@ -72,17 +72,34 @@ group.
 
 PRINCIPALS
 
-All ACLs require a principal. A principal is composed of two parts: the type
-and the name. Within Redpanda, only one type is supported, "User". The reason
-for the prefix is that a potential future authorizer may add support for
-authorizing by Group or anything else.
+All ACLs require a principal or a role. A principal comprises a user and type.
+Within Redpanda, only the "User" type is supported. The "User" prefix ensures
+that potential future authorizers can add support for authorization by Group
+or some other type.
```

Comment:
This is unclear to me. How does the user type ensure that future authorizers can add support for groups or other types?

Reply:
This is sort of a carryover from the Kafka API. The general idea is to namespace user principals so that other types of principals can be added in the future. Incidentally, that's exactly how roles work under the hood! (a role named "my_role" becomes "RedpandaRole:my_role")

Maybe worth noting that this phrasing is not new to this diff and appears in publicly available `rpk` help text currently. Which is just to say that it probably shouldn't be a release blocker, though I agree that the language is confusing 😕

Reply:
Missed the second half of your comment, yes, that's the thrust of it.